### ISSUE TITLE
App crashes on startup

### ISSUE THREAD

**Issue by Pingou** [realm/realm-js#5083]
### How frequently does the bug occur?

Sometimes

### Description

Since I updated my app some users are crashing on iOS.
Every time they open the app it immediately crashes.
I use the same realm version as I previously used, the only change is that I started from a new React Native project and copy pasted my old files. (and upgraded realm).
I had to stop the release, any help?
Users crashing are users who previously had a realm database from 11.0.0-rc.1 (although db was created with a previous version, don't know which).

### Stacktrace & log output

```shell
Thread 16 name:  com.facebook.react.JavaScript
Thread 16 Crashed:
0   libsystem_kernel.dylib        	       0x1c879cb38 __pthread_kill + 8
1   libsystem_pthread.dylib       	       0x20250b3bc pthread_kill + 268
2   libsystem_c.dylib             	       0x19bd0644c __abort + 128
3   libsystem_c.dylib             	       0x19bcae530 abort + 180
4   app                           	       0x101004970 realm::util::terminate+ 10176880 (char const*, char const*, long, std::initializer_list<realm::util::Printable>&&) + 0
5   app                           	       0x101004c58 realm::util::terminate_internal+ 10177624 (std::__1::basic_stringstream<char, std::__1::char_traits<char>, std::__1::allocator<char> >&) + 260
6   app                           	       0x101004e54 realm::util::terminate_with_info+ 10178132 (char const*, char const*, long, char const*, std::initializer_list<realm::util::Printable>&&) + 388
7   app                           	       0x100f1baf8 realm::GroupWriter::get_free_space+ 9222904 (unsigned long) + 0
8   app                           	       0x100f1b418 realm::GroupWriter::recreate_freelist+ 9221144 (unsigned long) + 1636
9   app                           	       0x100f1a0d4 realm::GroupWriter::write_group+ 9216212 () + 572
10  app                           	       0x100f01f80 realm::DB::low_level_commit+ 9117568 (unsigned long long, realm::Transaction&, bool) + 400
11  app                           	       0x100f01cfc realm::DB::do_commit+ 9116924 (realm::Transaction&, bool) + 140
12  app                           	       0x100fe20e4 realm::Transaction::commit_and_continue_as_read+ 10035428 (bool) + 88
13  app                           	       0x100cf7f2c realm::_impl::RealmCoordinator::commit_write+ 6979372 (realm::Realm&, bool) + 100
14  app                           	       0x100d53334 realm::Realm::commit_transaction+ 7353140 () + 168
15  app                           	       0x100c3bf5c realm::js::RealmClass<realm::js::realmjsi::Types>::write+ 6209372 (realm::js::JsiEnv, realm::js::JsiObj, realm::js::Arguments<realm::js::realmjsi::Types>&, realm::js::ReturnValue<realm::js::realmjsi::Types>&) + 244
16  app                           	       0x100bf2490 facebook::jsi::Value realm::js::wrap<&(realm::js::RealmClass<realm::js::realmjsi::Types>::write(realm::js::JsiEnv, realm::js::JsiObj, realm::js::Arguments<realm::js::realmjsi::Types>&, realm::js::ReturnValue<realm::js::realmjsi::Types>&))>+ 5907600 (facebook::jsi::Runtime&, facebook::jsi::Value const&, facebook::jsi::Value const*, unsigned long) + 192
17  app                           	       0x1009a2220 std::__1::function<facebook::jsi::Value (facebook::jsi::Runtime&, facebook::jsi::Value const&, facebook::jsi::Value const*, unsigned long)>::operator()(facebook::jsi::Runtime&, facebook::jsi::Value const&, facebook::jsi::Value const*, unsigned long) const + 3482144 (function.h:1182)
18  app                           	       0x1009a1f68 facebook::jsc::JSCRuntime::createFunctionFromHostFunction(facebook::jsi::PropNameID const&, unsigned int, std::__1::function<facebook::jsi::Value (facebook::jsi::Runtime&, facebook::jsi::Value const&, facebook::jsi::Value const*, unsigned long)>)::HostFunctionMetadata::call(OpaqueJSContext const*, OpaqueJSValue*, OpaqueJSValue*, unsigned long, OpaqueJSValue const* const*, OpaqueJSValue const**) + 3481448 (JSCRuntime.cpp:1172)
19  JavaScriptCore                	       0x19c1e9708 JSC::callJSNonFinalObjectCallbackObject(JSC::JSGlobalObject*, JSC::CallFrame*) + 416
20  JavaScriptCore                	       0x19ca2b7fc JSC::handleHostCall(JSC::JSGlobalObject*, JSC::CallFrame*, JSC::JSValue, JSC::CallLinkInfo*) + 544
21  JavaScriptCore                	       0x19ca5ede0 llint_link_call + 492
22  JavaScriptCore                	       0x19c16d41c llint_function_for_construct_arity_checkTagGateAfter + 160728
23  JavaScriptCore                	       0x19c169bd8 llint_function_for_construct_arity_checkTagGateAfter + 146324
24  JavaScriptCore                	       0x19c169bd8 llint_function_for_construct_arity_checkTagGateAfter + 146324
25  JavaScriptCore                	       0x19c169bd8 llint_function_for_construct_arity_checkTagGateAfter + 146324
26  JavaScriptCore                	       0x19c169bd8 llint_function_for_construct_arity_checkTagGateAfter + 146324
27  JavaScriptCore                	       0x19c169b20 llint_function_for_construct_arity_checkTagGateAfter + 146140
28  JavaScriptCore                	       0x19c169b20 llint_function_for_construct_arity_checkTagGateAfter + 146140
29  JavaScriptCore                	       0x19c169bd8 llint_function_for_construct_arity_checkTagGateAfter + 146324
30  JavaScriptCore                	       0x19c169bd8 llint_function_for_construct_arity_checkTagGateAfter + 146324
31  JavaScriptCore                	       0x19c169b20 llint_function_for_construct_arity_checkTagGateAfter + 146140
32  JavaScriptCore                	       0x19c169b20 llint_function_for_construct_arity_checkTagGateAfter + 146140
33  JavaScriptCore                	       0x19c16a9f4 llint_function_for_construct_arity_checkTagGateAfter + 149936
34  JavaScriptCore                	       0x19c16b120 llint_function_for_construct_arity_checkTagGateAfter + 151772
35  JavaScriptCore                	       0x19c169bd8 llint_function_for_construct_arity_checkTagGateAfter + 146324
36  JavaScriptCore                	       0x19c169bd8 llint_function_for_construct_arity_checkTagGateAfter + 146324
37  JavaScriptCore                	       0x19c169bd8 llint_function_for_construct_arity_checkTagGateAfter + 146324
38  JavaScriptCore                	       0x19c169bd8 llint_function_for_construct_arity_checkTagGateAfter + 146324
39  JavaScriptCore                	       0x19c169bd8 llint_function_for_construct_arity_checkTagGateAfter + 146324
40  JavaScriptCore                	       0x19c13fb10 vmEntryToJavaScriptTrampoline + 8
41  JavaScriptCore                	       0x19c93dad4 JSC::Interpreter::executeCall(JSC::JSGlobalObject*, JSC::JSObject*, JSC::CallData const&, JSC::JSValue, JSC::ArgList const&) + 536
42  JavaScriptCore                	       0x19cc876bc JSC::boundThisNoArgsFunctionCall(JSC::JSGlobalObject*, JSC::CallFrame*) + 496
43  JavaScriptCore                	       0x19c16d230 llint_function_for_construct_arity_checkTagGateAfter + 160236
44  JavaScriptCore                	       0x19c169bd8 llint_function_for_construct_arity_checkTagGateAfter + 146324
45  JavaScriptCore                	       0x19c169bd8 llint_function_for_construct_arity_checkTagGateAfter + 146324
46  JavaScriptCore                	       0x19c169bd8 llint_function_for_construct_arity_checkTagGateAfter + 146324
47  JavaScriptCore                	       0x19c169bd8 llint_function_for_construct_arity_checkTagGateAfter + 146324
48  JavaScriptCore                	       0x19c169bd8 llint_function_for_construct_arity_checkTagGateAfter + 146324
49  JavaScriptCore                	       0x19c16b120 llint_function_for_construct_arity_checkTagGateAfter + 151772
50  JavaScriptCore                	       0x19c169b20 llint_function_for_construct_arity_checkTagGateAfter + 146140
51  JavaScriptCore                	       0x19c169bd8 llint_function_for_construct_arity_checkTagGateAfter + 146324
52  JavaScriptCore                	       0x19c169b20 llint_function_for_construct_arity_checkTagGateAfter + 146140
53  JavaScriptCore                	       0x19c13fb10 vmEntryToJavaScriptTrampoline + 8
54  JavaScriptCore                	       0x19c93dad4 JSC::Interpreter::executeCall(JSC::JSGlobalObject*, JSC::JSObject*, JSC::CallData const&, JSC::JSValue, JSC::ArgList const&) + 536
55  JavaScriptCore                	       0x19cc876bc JSC::boundThisNoArgsFunctionCall(JSC::JSGlobalObject*, JSC::CallFrame*) + 496
56  JavaScriptCore                	       0x19c13fcc0 vmEntryToNative + 284
57  JavaScriptCore                	       0x19c93db24 JSC::Interpreter::executeCall(JSC::JSGlobalObject*, JSC::JSObject*, JSC::CallData const&, JSC::JSValue, JSC::ArgList const&) + 616
58  JavaScriptCore                	       0x19cbc548c JSC::profiledCall(JSC::JSGlobalObject*, JSC::ProfilingReason, JSC::JSValue, JSC::CallData const&, JSC::JSValue, JSC::ArgList const&) + 96
59  JavaScriptCore                	       0x19c1fcc44 JSObjectCallAsFunction + 520
60  app                           	       0x1009a04cc facebook::jsc::JSCRuntime::call(facebook::jsi::Function const&, facebook::jsi::Value const&, facebook::jsi::Value const*, unsigned long) + 3474636 (JSCRuntime.cpp:0)
61  app                           	       0x1009ac534 facebook::jsi::Value facebook::jsi::Function::call<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, facebook::jsi::Value>(facebook::jsi::Runtime&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, facebook::jsi::Value&&) const + 3523892 (jsi-inl.h:247)
62  app                           	       0x1009ac398 std::__1::__function::__func<facebook::react::JSIExecutor::callFunction(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, folly::dynamic const&)::$_4, std::__1::allocator<facebook::react::JSIExecutor::callFunction(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, folly::dynamic const&)::$_4>, void ()>::operator()() + 3523480 (function.h:352)
63  app                           	       0x1008dbec4 void std::__1::__invoke_void_return_wrapper<void, true>::__call<void (*&)(std::__1::function<void ()> const&, std::__1::function<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > ()>), std::__1::function<void ()> const&, std::__1::function<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > ()> >(void (*&)(std::__1::function<void ()> const&, std::__1::function<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > ()>), std::__1::function<void ()> const&, std::__1::function<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > ()>&&) + 2670276 (invoke.h:61)
64  app                           	       0x1009a9818 facebook::react::JSIExecutor::callFunction(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, folly::dynamic const&) + 3512344 (JSIExecutor.cpp:253)
65  app                           	       0x10099cb2c std::__1::__function::__func<facebook::react::NativeToJsBridge::runOnExecutorQueue(std::__1::function<void (facebook::react::JSExecutor*)>)::$_8, std::__1::allocator<facebook::react::NativeToJsBridge::runOnExecutorQueue(std::__1::function<void (facebook::react::JSExecutor*)>)::$_8>, void ()>::operator()() + 3459884 (function.h:352)
66  app                           	       0x1008f80c8 facebook::react::tryAndReturnError(std::__1::function<void ()> const&) + 2785480 (function.h:0)
67  app                           	       0x100903188 facebook::react::RCTMessageThread::tryFunc(std::__1::function<void ()> const&) + 2830728 (RCTMessageThread.mm:69)
68  app                           	       0x100902f3c invocation function for block in facebook::react::RCTMessageThread::runAsync(std::__1::function<void ()>) + 2830140 (RCTMessageThread.mm:46)
69  CoreFoundation                	       0x190b72504 __CFRUNLOOP_IS_CALLING_OUT_TO_A_BLOCK__ + 28
70  CoreFoundation                	       0x190b73460 __CFRunLoopDoBlocks + 412
71  CoreFoundation                	       0x190b0afb0 __CFRunLoopRun + 848
72  CoreFoundation                	       0x190b1eb0c CFRunLoopRunSpecific + 600
73  app                           	       0x1008ee2bc +[RCTCxxBridge runRunLoop] + 2745020 (RCTCxxBridge.mm:0)
74  Foundation                    	       0x1923613dc __NSThread__start__ + 808
75  libsystem_pthread.dylib       	       0x2025059ac _pthread_start + 148
76  libsystem_pthread.dylib       	       0x202504e68 thread_start + 8
```

### Can you reproduce the bug?

Not yet

### Version

11.0.0-rc.1

RN: 0.70
Hermes: no

### What SDK flavour are you using?

Local Database only

### Are you using encryption?

No, not using encryption

### Platform OS and version(s)

iOS 15, iOS 16

### Build environment

Which debugger for React Native: ..

**Comment by kneth** [comment]
@Pingou Thank you for reporting. We are currently [investigating a series of crashes](https://github.com/realm/realm-core/pull/5993) in our SDKs supporting iOS, and we hope to release a fix soon.

**Comment by kneth** [comment]
The fix was released in version 11.2.0.
